src/services/gnosis/error.ts

```diff
@@ -63,30 +98,52 @@ export default class OperatorError extends Error {
       console.error(
         'Error handling a 400 error. Likely a problem deserialising the JSON response'
       );
-      return OperatorError.apiErrorDetails.UNHANDLED_ERROR;
+      return _mapActionToErrorDetail(action);
     }
   }
-  static getErrorForUnsuccessfulPostOrder(response: AxiosResponse) {
+  static getErrorFromStatusCode(
+    response: AxiosResponse,
+    action: 'create' | 'delete'
+  ) {
     switch (response.status) {
       case 400:
-        return this.getErrorMessage(response);
+      case 404:
+        return this.getErrorMessage(response.data, action);
 
       case 403:
-        return 'The order cannot be accepted. Your account is deny-listed.';
+        return `The order cannot be ${
+          action === 'create' ? 'accepted' : 'cancelled'
+        }. Your account is deny-listed.`;
 
       case 429:
-        return 'The order cannot be accepted. Too many order placements. Please, retry in a minute';
+        return `The order cannot be ${
+          action === 'create'
+            ? 'accepted. Too many order placements'
+            : 'cancelled. Too many order cancellations'
+        }. Please, retry in a minute`;
 
       case 500:
       default:
-        return 'Error adding an order';
+        console.error(
+          `[OperatorError::getErrorFromStatusCode] Error ${
+            action === 'create' ? 'creating' : 'cancelling'
+          } the order, status code:`,
+          response.status || 'unknown'
+        );
+        return `Error ${
+          action === 'create' ? 'creating' : 'cancelling'
+        } the order`;
     }
   }
-  constructor(apiError: ApiError) {
+  constructor(apiError: ApiErrorObject) {
     super(apiError.description);
 
     this.type = apiError.errorType;
     this.description = apiError.description;
-    this.message = OperatorError.apiErrorDetails[apiError.errorType];
+    this.message = ApiErrorCodeDetails[apiError.errorType];
   }
 }
+
+export function isValidOperatorError(error: any): error is OperatorError {
+  return error instanceof OperatorError;
+}
```

Comment:
This is mostly a port from an updated version of the file from CowSwap that handles operator errors.